### PR TITLE
rework admin weather commands

### PR DESCRIPTION
**Pull request by razaqq**

### CODE REVIEW

**Comment by compujuckel** [comment]
IMHO it would be nice instead of a new command to make `/setcspweather` accept weather names, like `/setcspweather HeavyRain 120`. Maybe that's even possible with Qmmands by just passing the `upcoming` parameter as `WeatherFxType` instead of `int`

**Comment by razaqq** [comment]
> IMHO it would be nice instead of a new command to make `/setcspweather` accept weather names, like `/setcspweather HeavyRain 120`. Maybe that's even possible with Qmmands by just passing the `upcoming` parameter as `WeatherFxType` instead of `int`

should be no problem with reflection, i would still add a command to list all weathers tho, but by string and not int

**Comment by razaqq** [comment]
done

File: AssettoServer/Commands/Modules/AdminModule.cs [hunk 1]
```diff
@@ -108,11 +110,29 @@ public void SetWeather(int weatherId)
...
+        Context.Client.SendPacket(new ChatMessage { SessionId = 255, Message = "Available weathers:" });
+        foreach (WeatherFxType weather in weathers)
+        {
+            Context.Client.SendPacket(new ChatMessage { SessionId = 255, Message = $" - {weather}" });
+        }
+    }
+
     [Command("setcspweather")]
-    public void SetCspWeather(int upcoming, int duration)
+    public void SetCspWeather(string upcomingWeatherName, int duration)
```
Comment (compujuckel) [inline AssettoServer/Commands/Modules/AdminModule.cs:125]: you can change the parameter type from `string` to `WeatherFxType` and Qmmands will handle the enum parsing for you

File: AssettoServer/Commands/Modules/AdminModule.cs [hunk 2]
```diff
@@ -108,11 +110,29 @@ public void SetWeather(int weatherId)
         }
     }
 
+    [Command("cspweather")]
+    public void CspWeather()
+    {
+        List<WeatherFxType> weathers = Enum.GetValues<WeatherFxType>().ToList();
```
Comment (compujuckel) [inline AssettoServer/Commands/Modules/AdminModule.cs:116]: really just a nitpick but you could remove the `ToList` and save one list allocation

Comment (razaqq) [reply AssettoServer/Commands/Modules/AdminModule.cs]: yeah your right, dont use c# very often (mostly c/cpp/rust) so i might miss some details like this
fixed

File: AssettoServer/Commands/Modules/AdminModule.cs (hunk 1 above)
Comment (compujuckel) [reply AssettoServer/Commands/Modules/AdminModule.cs]: hm you're right, the custom message with a reference to the /cspweather command is much better here. Feel free to re-add that part

**Pull request merged**